src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    public Node<T> first;
+    public Node<T> last;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node previous;
+        private Node next;
+
+        public Node(Node previous, T element, Node next) {
+            this.previous = previous;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
+    private Node getNodeByIndex(int index) {
+        if (index == 0) {
+            return first;
+        }
+        if (index == size - 1) {
+            return last;
+        }
+        Node result = first;
+        for (int i = 1; i <= index; i++) {
+            result = result.next;
+        }
+        return result;
+    }
+
+    private Node getNodeByElement(T t) {
```

Comment:
@TarasHryts what about generics?)
```suggestion
    private Node<T> getNodeByElement(T t) {
```